### PR TITLE
Update superlu-dist package.py to support latest release v6.0.0

### PR DESCRIPTION
**Pull request by gchavez2**
Update spack script of superlu-dist to use CMake build

### CODE REVIEW

**Comment by balay** [comment]
@davydden, how do we implement:

if version >5.4: use cmakebuild - else: use (old) make.inc build

@gchavez2, for blas fix - please cherry-pick 547f07189c8c368f3365f611d57933a2dcbed7f4 into your PR branch

**Comment by healther** [comment]
@balay I overwrote the install phases that `CmakePackage` implements and just reimplemented the `install` method to do the "old" build.

**Comment by davydden** [comment]
@gchavez2 @balay 

> how do we implement: if version >5.4: use cmakebuild - else: use (old) make.inc build

look at https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/superlu/package.py which combines Make and Cmake

**Comment by balay** [comment]
@davydden , @healther, thanks for the pointers.

The cmake build works for versions 5.0.0+ higher. And @xiaoyeli indicated we should ignore older versions of superlu-dist.

So is 1ef218eff9e39815c3428dc427a881fbfcaf1304 ok?

**Comment by balay** [comment]
@davydden Thanks! I guess if there is a real need to support older versions - we could add it [using the mentioned examples]

@gchavez2 can you cherry-pick the above 2 commits into your PR branch - so that the PR gets updated? [or grab the patches - and apply them.]

**Comment by gchavez2** [comment]
I updated the spack script by specifying lapack_blas and DCMAKE_INSTALL_LIBDIR
and I disabled support for superlu-dist before v5, double checking with @xiaoyeli

Thanks @balay @davydden and @healther

**Comment by balay** [comment]
@gchavez2, you missed:
```
-    url      = "https://github.com/xiaoyeli/superlu_dist/archive/v6.0.0.tar.gz"
+    url      = "http://crd-legacy.lbl.gov/~xiaoye/SuperLU/superlu_dist_6.0.0.tar.gz"
<snip>
-    version('6.0.0', '2e3ce927fa5786470dacbdf8c41afb08')
+    version('6.0.0', '4e57072c3be26809d271bf1adc15c834')
```
Either this - or recompute the md5sums for versions 5.0.0 to 5.4 for github url

File: var/spack/repos/builtin/packages/superlu-dist/package.py [hunk 1]
```diff
@@ -22,21 +22,20 @@
...
     nonsymmetric systems of linear equations on high performance machines."""
 
     homepage = "http://crd-legacy.lbl.gov/~xiaoye/SuperLU/"
-    url      = "http://crd-legacy.lbl.gov/~xiaoye/SuperLU/superlu_dist_4.1.tar.gz"
+    url      = "https://github.com/xiaoyeli/superlu_dist/archive/v6.0.0.tar.gz"
     git      = "https://github.com/xiaoyeli/superlu_dist.git"
 
     version('develop', branch='master')
     version('xsdk-0.2.0', tag='xsdk-0.2.0')
+    version('6.0.0', '2e3ce927fa5786470dacbdf8c41afb08')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/superlu-dist/package.py]: oh, please use `sha256`, something like 
```
sha256='8311ab1a57f142f8057508d4c360e086fcea32f020b4181fed88fee72554351c'
```

you can do `spack fetch superlu-dist@x.y.z` to check/update others.

Comment (balay) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: If github URLs are preferable - we could just use git tags?

```
+    version('6.0.0', 'tag=v6.0.0')
+    version('5.4.0', 'tag=v5.4.0')
```

Comment (davydden) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: i don't think tags are checksummed in Spack, so IMO it's better to use tarbals.

Comment (balay) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: well - my thought here is -  git keeps track of checksums - so spack doesn't need to do that  extra work.. [when using git tags]

Comment (davydden) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: i was thinking about the case when tags are moved forcefully after creation. Of course, that's not the case for this package.

Comment (healther) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: You could add the commit hash rather than the tag. The commit is "secure", the tag isn't

Comment (davydden) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: sounds good to me.

Comment (balay) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: @healther , yeah commit is another way to do it.

@gchavez2 hasn't clarified the reason for preferring git tarball URLs - so for now - I'll just merge this in - and add a new PR with sha256 sums - and worry about alternate formats later.
cc:  @xiaoyeli

Comment (balay) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: https://github.com/spack/spack/pull/9499

**Comment by citibeth** [comment]
@gchavez2 @balay @scheibelp 

**Please**.... we *cannot* simply remove old versions of a package just because we added a new version.

@balay... Removing old versions should never have been approved or merged, without first *carefully* checking whether people use the old versions that are being removed.  In this case such a check would have turned up that `petsc/package.py` uses version 4.3.  Removing old versions broke the `petsc` build in mysterious ways, leading to wasted time and spurious / mysterious bug reports (see #12938).

Can you please submit and merge a PR to restore the old versions that were removed here?  Thank you!

**Comment by sethrj** [comment]
Ugh ditto @citibeth -- I spent the last half hour figuring out why `petsc@3.6.4` was failing to build. The second half of that bug is that somehow the petsc package's `depends_on('superlu-dist@:4.3~int64', when='@3.4.4:3.6.4+superlu-dist+mpi~int64')`
caused the concretizer to accept `^superlu-dist@xsdk-0.2.0` as a valid version. See new issue https://github.com/spack/spack/issues/17145 .

cc also @gchavez2 @scheibelp

**Pull request merged**